### PR TITLE
fix: batch embedding queries, fix get client in tests

### LINKED ISSUE #2905: Bedrock Embeddings ThrottlingException
When attempting to generate a vector store using the ```fromDocuments``` method  on approximately 10,000 documents (specifically on the ```HNSWLib``` class) , it reaches the rate limit quickly. Most likely due to the ```BedrockEmbeddings.embedDocuments``` method just being a simple ```Promise.all``` on every document at once.

### PR DESCRIPTION
**Pull request by bracesproul**
Closes #2905 
Batch requests for aws bedrock embeddings

### CODE REVIEW

File: langchain/src/embeddings/bedrock.ts [hunk 1]
```diff
@@ -53,6 +55,7 @@ export class BedrockEmbeddings
       });
   }
 
+  /** @deprecated */
```
Comment (bracesproul) [inline langchain/src/embeddings/bedrock.ts:58]: Side question, could this method be deleted instead or would that not maintain backcompat?

File: langchain/src/embeddings/bedrock.ts [hunk 2]
```diff
@@ -93,13 +115,23 @@ export class BedrockEmbeddings
...
+   * @param documents Array of texts for which to generate embeddings.
    * @returns Promise that resolves to a 2D array of embeddings for each input document.
    */
-  embedDocuments(documents: string[]): Promise<number[][]> {
-    return Promise.all(documents.map((document) => this._embedText(document)));
+  async embedDocuments(documents: string[]): Promise<number[][]> {
+    const batches = chunkArray(documents, this.batchSize);
+    const embeddings: number[][] = [];
+
+    for (const batch of batches) {
```
Comment (jacoblee93) [inline langchain/src/embeddings/bedrock.ts:128]: You don't actually have to batch here - just `Promise.all` everything. The `caller` will take care of concurrency.

Comment (jacoblee93) [reply langchain/src/embeddings/bedrock.ts:128]: https://github.com/langchain-ai/langchainjs/blob/main/langchain/src/embeddings/openai.ts#L159

**Comment by jacoblee93** [comment]
Awesome, thank you!

**Pull request merged**